lib/plugins/Label_22_OFF.test.ts

```diff
@@ -0,0 +1,112 @@
+    const decodeResult = plugin.decode({ text: text });
+    expect(decodeResult.decoded).toBe(true);
+    expect(decodeResult.decoder.decodeLevel).toBe('partial');
+    expect(decodeResult.raw.flight_number).toBe('YX3661');
+    expect(decodeResult.raw.departure_day_of_month).toBe(25);
+    expect(decodeResult.raw.arrival_day_of_month).toBe(25);
+    expect(decodeResult.raw.time_of_day).toBe(61920);
+    expect(decodeResult.raw.off_time).toBe(61927);
+    expect(decodeResult.raw.departure_icao).toBe('KIAD');
+    expect(decodeResult.raw.arrival_icao).toBe('KPWM');
+    expect(decodeResult.formatted.items.length).toBe(7);
+    expect(decodeResult.formatted.items[0].label).toBe('Flight Number');
+    expect(decodeResult.formatted.items[0].value).toBe('YX3661');
+    expect(decodeResult.formatted.items[1].label).toBe('Departure Day');
+    expect(decodeResult.formatted.items[1].value).toBe('25');
+    expect(decodeResult.formatted.items[2].label).toBe('Arrival Day');
+    expect(decodeResult.formatted.items[2].value).toBe('25');
+    expect(decodeResult.formatted.items[3].label).toBe('Message Timestamp');
+    expect(decodeResult.formatted.items[3].value).toBe('17:12:00');
+    expect(decodeResult.formatted.items[4].label).toBe('Origin');
+    expect(decodeResult.formatted.items[4].value).toBe('KIAD');
+    expect(decodeResult.formatted.items[5].label).toBe('Destination');
+    expect(decodeResult.formatted.items[5].value).toBe('KPWM');
+    expect(decodeResult.formatted.items[6].label).toBe('Takeoff Time');
+    expect(decodeResult.formatted.items[6].value).toBe('17:12:07');
+    expect(decodeResult.remaining.text).toBe('  92');
+  });
+
+  test('decodes variant 2', () => {
+    const text = 'OFF02XA0000/N38568 W077261251152KIADEPRZ1152****1958'
+    const decodeResult = plugin.decode({ text: text });
+    expect(decodeResult.decoded).toBe(true);
+    expect(decodeResult.decoder.decodeLevel).toBe('partial');
+    expect(decodeResult.raw.position.latitude).toBe(38.946666666666665);
+    expect(decodeResult.raw.position.longitude).toBe(-77.435);
+    expect(decodeResult.raw.day_of_month).toBe(25);
+    expect(decodeResult.raw.time_of_day).toBe(42720);
+    expect(decodeResult.raw.off_time).toBe(42720);
+    expect(decodeResult.raw.departure_icao).toBe('KIAD');
+    expect(decodeResult.raw.arrival_icao).toBe('EPRZ');
+    expect(decodeResult.formatted.items.length).toBe(8);
+    expect(decodeResult.formatted.items[0].label).toBe('Flight Number');
+    expect(decodeResult.formatted.items[0].value).toBe('XA0000');
+    expect(decodeResult.formatted.items[1].label).toBe('Aircraft Position');
+    expect(decodeResult.formatted.items[1].value).toBe('38.947 N, 77.435 W');
+    expect(decodeResult.formatted.items[2].label).toBe('Day of Month');
+    expect(decodeResult.formatted.items[2].value).toBe('25');
+    expect(decodeResult.formatted.items[3].label).toBe('Message Timestamp');
+    expect(decodeResult.formatted.items[3].value).toBe('11:52:00');
```

Comment:
in this variant, both the timestamp and takeoff time have minute resolution - i could be parsing backwards but it probably doesn't matter